core/schema/envfile.go

```diff
@@ -0,0 +1,90 @@
+package schema
+
+import (
+	"context"
+
+	"github.com/dagger/dagger/core"
+	"github.com/dagger/dagger/dagql"
+)
+
+type envfileSchema struct{}
+
+var _ SchemaResolvers = &envfileSchema{}
+
+func (s envfileSchema) Install(srv *dagql.Server) {
+	dagql.Fields[*core.Query]{
+		dagql.Func("envFile", s.newEnvFile).
+			Doc("Initialize an environment file"),
+	}.Install(srv)
+
+	dagql.Fields[*core.EnvFile]{
+		dagql.Func("variables", s.variables).
+			Doc("Return all variables"),
+		dagql.Func("withVariable", s.withVariable).
+			Doc("Add a variable").
+			Args(
+				dagql.Arg("name").
+					Doc("Variable name"),
+				dagql.Arg("value").
+					Doc("Variable value"),
+			),
```

Comment:
tiny nit: any reason to not put `Doc` on the same line as `Arg`?

Reply:
I find it more readable but obviously it's a highly subjective opinion 😛 Happy to conform to whatever the convention is.